src/overlays/actors/ovl_En_Knight/z_en_knight.c

```diff
@@ -205,237 +380,4015 @@ ActorInit En_Knight_InitVars = {
+        ShrinkWindow_Letterbox_SetSizeTarget(27);
+        Play_SetCameraAtEye(play, this->csCamId, &this->csCamAt, &this->csCamEye);
+        Play_SetCameraFov(play, this->csCamId, this->csCamFov);
+    }
+}
+
+void EnKnight_SetupKnightCaptainsHatCS(EnKnight* this, PlayState* play) {
+    if (this->actionFunc != EnKnight_FallOver && this->actionFunc != EnKnight_Die &&
+        this->actionFunc != EnKnight_Dead) {
+        Animation_MorphToLoop(&this->skelAnime, &gKnightIdleAnim, -5.0f);
+        this->actionFunc = EnKnight_KnightCaptainsHatCS;
+    }
+}
+
+void EnKnight_KnightCaptainsHatCS(EnKnight* this, PlayState* play) {
+    Vec3f offsetFromPlayer;
+    Player* player = GET_PLAYER(play);
+
+    SkelAnime_Update(&this->skelAnime);
+    this->neckPitchTarget = 0;
+    this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.speed = 0.0f;
+
+    if (sIgosInstance->subAction == KNIGHT_SUB_ACTION_CAPTAINS_HAT_CS_IGOS_SITTING) {
+        // If Igos is still sitting, position the other two at specific positions behind the player
+        if (this == sYaseInstance) {
+            Matrix_RotateYS(sIgosInstance->actor.shape.rot.y + 0x1400, MTXMODE_NEW);
+        } else {
+            Matrix_RotateYS(sIgosInstance->actor.shape.rot.y - 0x1400, MTXMODE_NEW);
+        }
+
+        Matrix_MultVecZ((KREG(19) + 350.0f) - 220.0f, &offsetFromPlayer);
+        this->actor.world.pos.x = player->actor.world.pos.x + offsetFromPlayer.x;
+        this->actor.world.pos.z = player->actor.world.pos.z + offsetFromPlayer.z;
+    }
+
+    if (sIgosInstance->actionFunc != EnKnight_CaptainsHatCS) {
+        EnKnight_SetupWait(this, play);
+    }
+}
+
+void EnKnight_SetupCaptainsHatCS(EnKnight* this, PlayState* play) {
+    this->csTimer = 0;
+    this->csState = KNIGHT_CS_2_STATE_0;
+    this->csStepValue = 0.0f;
+
+    SEQCMD_STOP_SEQUENCE(SEQ_PLAYER_BGM_MAIN, 10);
+
+    if (this->actionFunc != EnKnight_IgosSitting) {
+        this->subAction = KNIGHT_SUB_ACTION_CAPTAINS_HAT_CS_IGOS_STANDING;
+        this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    } else {
+        this->subAction = KNIGHT_SUB_ACTION_CAPTAINS_HAT_CS_IGOS_SITTING;
+    }
+
+    EnKnight_SetupKnightCaptainsHatCS(sYaseInstance, play);
+    EnKnight_SetupKnightCaptainsHatCS(sDebuInstance, play);
+    this->prevActionFunc = this->actionFunc;
+    this->actionFunc = EnKnight_CaptainsHatCS;
+}
+
+void EnKnight_CaptainsHatCS(EnKnight* this, PlayState* play) {
+    Vec3f posOffset;
+    Player* player = GET_PLAYER(play);
+
+    this->neckPitchTarget = 0;
+    this->actor.speed = 0.0f;
+    this->csTimer++;
+    this->animTranslationX = 0.0f;
+    this->animTranslationZ = 0.0f;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    switch (this->csState) {
+        case KNIGHT_CS_2_STATE_0:
+            if (CutsceneManager_GetCurrentCsId() != -1) {
```

Comment:
```suggestion
            if (CutsceneManager_GetCurrentCsId() != CS_ID_NONE) {
```
If you could do it for the other places as well that would be great :)